pkg/build/strategies/layered/layered.go

```diff
@@ -41,10 +42,13 @@ type Layered struct {
 }
 
 // New creates a Layered builder.
-func New(config *api.Config, scripts build.ScriptsHandler, overrides build.Overrides) (*Layered, error) {
+func New(config *api.Config, scripts build.ScriptsHandler, overrides build.Overrides) (*Layered, []*api.BuildStepInfo, error) {
```

Comment:
i understand what you mean. though being clear about it, this PR doesn't add anything useful for s2i itself (or it might be of use in the future, or a follow up PR), but for origin. in openshift, i don't want to compare errors, i want to compare constants when i try to find out what went wrong. Also, whenever we add the timing of functions, it should be easier to transport the information up the stack and into openshift.
openshift uses the Build() interface, that's why the second value is ignored in s2i. the build() returns a *result, that's where i need the info.